app/screens/settings/display_settings/display_settings.js

```diff
@@ -79,7 +78,7 @@ export default class DisplaySettings extends PureComponent {
         const style = getStyleSheet(theme);
 
         let clockDisplayModal;
-        if (Platform.OS === 'android') {
+        if (Platform.OS === 'android' && showClockDisplaySettings) {
```

Comment:
I'm 1/5 that we shouldn't be managing visibility of the model outside the `showModal` visibility prop.

I get the fact that we avoid the `ClockDisplay` component entirely via the `&&` check above, but this makes `showModal` redundant, and takes away the benefit of modals being instantly available when invoked because of a quicker re-render, instead of a rebuild of the component tree from this parent downwards. At this point then, I wonder if `ClockDisplay` should even be defined as a "modal".

Reply:
Yes, this can be reverted.. is totally true that the overhead is more than just the overdraw

Reply:
done